node/app/query.go

```diff
@@ -196,63 +163,38 @@ func HandleUtxoQuery(app Application, message []byte) []byte {
 	if len(parts) > 1 {
 		name = parts[1]
 	}
-	result := UtxoInfo(app, name)
-	log.Debug("Returning", "result", string(result))
-	return result
+	return UtxoInfo(app, name)
 }
 
-func UtxoInfo(app Application, name string) []byte {
-	buffer := ""
+func UtxoInfo(app Application, name string) interface{} {
 	if name == "" {
 		entries := app.Utxo.FindAll()
-		for key, value := range entries {
-			account, errs := app.Accounts.FindKey([]byte(key))
-			if errs != err.SUCCESS {
-				log.Fatal("Accounts", "err", errs, "key", key)
-			}
-
-			var name string
-			if account == nil {
-				name = fmt.Sprintf("%X", key)
-			} else {
-				name = account.Name() + "@" + fmt.Sprintf("%X", key)
-			}
-
-			if value != nil {
-				buffer += name + ":" + value.AsString() + ", "
-			} else {
-				buffer += name + ":EMPTY, "
-			}
-		}
-	} else {
-		value := app.Utxo.Find(data.DatabaseKey(name))
-		buffer += name + ":" + value.AsString()
+		return entries
+
 	}
-	return []byte(buffer)
+	value := app.Utxo.Get(data.DatabaseKey(name))
+	return value
 }
 
 // Get the balancd for an account
 func GetBalance(app Application, account id.Account) string {
-	result := app.Utxo.Find(account.AccountKey())
-	if result == nil {
-		log.Warn("Balance Not Found", "key", account.AccountKey())
-		return " [missing]"
+	if account.Chain() != data.ONELEDGER {
+		return ""
 	}
 
+	result := app.Utxo.Get(account.AccountKey())
+	if result == nil {
+		return "[missing]"
+	}
 	return result.AsString()
 }
 
-// Return a nicely formatted error message
-func HandleError(text string, path string, massage []byte) []byte {
-	return []byte("Invalid Query")
-}
-
-func HandleVersionQuery(app Application, message []byte) []byte {
-	return []byte(version.Current.String())
+func HandleVersionQuery(app Application, message []byte) interface{} {
+	return version.Current.String()
 }
 
 // Get the account information for a given user
-func HandleBalanceQuery(app Application, message []byte) []byte {
+func HandleBalanceQuery(app Application, message []byte) interface{} {
```

Comment:
Why change all []byte to interface{}? Isn't []byte more reliable?

Reply:
Interface preserves the Go types. Byte arrays flatten the data (through techniques like serialization). The handlers now pass up their native responses, it is flattened once at the top (so you no longer have to worry about serialization or hex or anything in the actual queries).